File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task120_CLAUDE_claude-3-5-sonnet-20240620.java

import java.io.*;
import javax.servlet.*;
import javax.servlet.http.*;

public class DisplayUserInput extends HttpServlet {
    public void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        
        String userInput = request.getParameter("userInput");
        
        out.println("<html><body>");
        out.println("<h2>User Input:</h2>");
        out.println("<p>" + userInput + "</p>");
        out.println("</body></html>");
    }
    
    public void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        
        out.println("<html><body>");
        out.println("<h2>Enter your input:</h2>");
        out.println("<form method=\'post\'>");
        out.println("<input type=\'text\' name=\'userInput\'>");
        out.println("<input type=\'submit\' value=\'Submit\'>");
        out.println("</form>");
        out.println("</body></html>");
    }
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task120_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <string>
#include "crow.h"

int main() {
    crow::SimpleApp app;

    CROW_ROUTE(app, "/")
    ([](const crow::request& req) {
        std::string user_input = "";
        if (req.method == crow::HTTPMethod::Post) {
            user_input = req.get_body_params().get("user_input");
        }

        std::string html = R"(\n            <!DOCTYPE html>\n            <html>\n            <body>\n                <form method="post">\n                    <input type="text" name="user_input">\n                    <input type="submit" value="Submit">\n                </form>\n                <p>)" + user_input + R"(</p>\n            </body>\n            </html>\n        )";

        return crow::response(html);
    });

    app.port(8080).run();
    return 0;
}
